packages/flutter/lib/src/rendering/object.dart

```diff
@@ -965,6 +965,9 @@ class PipelineOwner {
   bool get debugDoingLayout => _debugDoingLayout;
   bool _debugDoingLayout = false;
 
+  // This is a method instead of a closure for optimization.
```

Comment:
If we have a benchmark for this, we should link it here. Otherwise, future readers/refactor-ers will have a hard time understanding whether this is safe to remove or change.

Alternatively, we could consider having some kind of lint for this (similar to how we "lint" against using sync*/async* in the framework) - although I'm not sure how challenging that would be to write.

Reply:
I wonder if we could have some way to track total number of allocations and have that turned on for something like transitions_perf.